legend-depot-core-services/src/main/java/org/finos/legend/depot/services/projects/ProjectsServiceImpl.java

```diff
@@ -127,6 +127,19 @@ public Optional<StoreProjectVersionData> find(String groupId, String artifactId,
         return projectsVersions.find(groupId, artifactId, versionId);
     }
 
+    @Override
+    public String resolveAliasesAndCheckVersionExists(String groupId, String artifactId, String versionId)
+    {
+        String version = versionId;
+        Optional<StoreProjectVersionData> projectVersion = this.find(groupId, artifactId, version);
+        if (projectVersion.isPresent())
+        {
+            version = projectVersion.get().getVersionId();
+        }
+        checkExists(groupId,artifactId,version);
```

Comment:
do we need to call check exist ?  if we have called  find we already have check for the version ,we just need to throw an exception if we could not find any.. the check exists is not necessary any more.
also can the checkexist be removed then? is it used anywher lese?